api/services/TokenAuth.js

```diff
@@ -1,21 +1,29 @@
-var TokenAuth = module.exports = {
+const crypto = require('crypto')
+const Promise = require('bluebird')
 
-  isValid: function(token) {
-    if (!process.env.KISS_AUTH_TOKEN) return false;
-    return token == process.env.KISS_AUTH_TOKEN;
-  },
+var TokenAuth = module.exports = {
 
-  setAuthenticated: function(res) {
-    res.locals.tokenAuthenticated = true;
+  generateToken: function() {
+    const randomBytes = Promise.promisify(crypto.randomBytes)
+    return randomBytes(24).then(buffer => buffer.toString('hex'))
   },
 
-  isAuthenticated: function(res) {
-    return !!res.locals.tokenAuthenticated;
+  checkAndSetAuthenticated: function(req) {
+    req.body = req.body || {}
+    const token = req.body.token || req.query.token || req.headers['x-access-token']
```

Comment:
let's call the param `access_token` or `auth_token` throughout. we already use the name `token` for other purposes, e.g. in `InvitationController`.